modules/miner/miner.go

```diff
@@ -16,9 +16,15 @@ const (
 
 	// headerForWorkMemory is the number of previous calls to 'headerForWork'
 	// that are remembered.
-	headerForWorkMemory = 50
+	headerForWorkMemory = 20
```

Comment:
I believe that this is too low. I was thinking X000.

Ideally, the total amount of memeory in use would be less than 50mb. I guess that gives you room for 20 blocks, and then if you do 100 headers per block you get to 2000 headers total in memory.

I think that's good enough...